src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -50,120 +51,108 @@ public Result charTyped(char charTyped,
                             @NotNull Project project,
                             @NotNull Editor editor,
                             @NotNull PsiFile file) {
-        LSPFormattingFeature formattingFeature = getFormattingFeature(file);
-        if (formattingFeature != null) {
-            boolean rangeFormattingSupported = formattingFeature.isRangeFormattingSupported(file);
+        // Gather all of the relevant client configuration
+        Ref<Boolean> rangeFormattingSupportedRef = Ref.create(false);
+        ClientSideOnTypeFormattingSettings onTypeFormattingSettings = new ClientSideOnTypeFormattingSettings();
+        LanguageServiceAccessor.getInstance(project).processLanguageServers(
+                file,
+                ls -> {
+                    if (ls.getServerStatus() == ServerStatus.started) {
```

Comment:
I think this test should be done in the processLanguageServers to be compliant with hasAny.

Reply:
Done.